pcsx2/R5900OpcodeImpl.cpp

```diff
@@ -982,8 +982,53 @@ void SYSCALL()
 		case Syscall::SetVTLBRefillHandler:
 			DevCon.Warning("A tlb refill handler is set. New handler %x", (u32*)PSM(cpuRegs.GPR.n.a1.UL[0]));
 			break;
-
-
+		case Syscall::StartThread:
+		case Syscall::ChangeThreadPriority:
+		{
+			if (CurrentBiosInformation.threadListAddr == 0)
+			{
+				u32 offset = 0x0;
+				// Suprisingly not that slow :)
+				while (offset < 0x5000) // I find that the instructions are in between 0x4000 -> 0x5000
+				{
+					// memRead128 only works with aligned addresses on linux
```

Comment:
memRead128 is aligned on *all platforms* and may break at any time when used unaligned on Windows (especially likely if we ever start allowing people to build with clang, but could also happen in the future with a compiler upgrade)

memRead64 is also aligned, and you're only getting away with it because there's no such thing as an aligned 64-bit load in x86

Reply:
Personally would recommend it

Technically if any of your reads cross page boundaries that aren't mapped in order (doubt it would actually happen as I think most of our pages are mapped pretty much 1-1) you'd be reading the wrong values even if you don't get an alignment exception